tests/ci/azure_pipeline_test/dsvm_nightly_linux_cpu.yml

```diff
@@ -45,22 +45,11 @@ jobs:
     parameters:
       conda_env: nightly_reco_base
 
-  - script: |
-      . /anaconda/etc/profile.d/conda.sh && \
-      conda activate nightly_reco_base && \
-      echo "Smoke tests" && \
-      pytest tests/smoke --durations 0 -m "smoke and not spark and not gpu" --junitxml=reports/test-smoke.xml && \
-      echo "Integration tests" && \
-      pytest tests/integration --durations 0 -m "integration and not spark and not gpu" --junitxml=reports/test-integration.xml && \
-      conda deactivate    
-    displayName: 'Run Tests'
-
-  - task: PublishTestResults@2
-    displayName: 'Publish Test Results '
-    inputs:
-      testResultsFiles: '**/test-*.xml'
-      failTaskOnFailedTests: true
-    condition: succeededOrFailed()
+  - template: steps/conda_pytest_linux.yml
```

Comment:
I'm trying to understand the workflow, this file will call tests/ci/azure_pipeline_test/steps/conda_pytest_linux.yml, right?
Two questions:
1) what does coalesce do?
2) are `PYSPARK_PYTHON` and `PYSPARK_DRIVER_PYTHON` always set, even if they are not in the spark environment?

Reply:
1. it fills in the first string that is not empty. I am using it to minimize duplicated inputs. in particular, the notebooks has a strange mix of params, where it most cases i need to fill in "unit", but in other cases "notebook"
2. yes, as they are just env vars, (and i'm still working on my devops jujitsu to do conditional addition of params), i decided to just set them.

Reply:
foot in mouth: i figured out how to only insert the parmas for spark.